server/src/main/java/org/elasticsearch/action/datastreams/GetDataStreamAction.java

```diff
@@ -242,7 +285,27 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params, @Nulla
                     .startObject()
                     .field(DataStream.NAME_FIELD.getPreferredName(), DataStream.TIMESTAMP_FIELD_NAME)
                     .endObject();
-                builder.xContentList(DataStream.INDICES_FIELD.getPreferredName(), dataStream.getIndices());
+
+                builder.field(DataStream.INDICES_FIELD.getPreferredName());
+                if (dataStream.getIndices() == null) {
+                    builder.nullValue();
+                } else {
+                    builder.startArray();
+                    for (Index index : dataStream.getIndices()) {
+                        builder.startObject();
+                        index.toXContentFragment(builder);
+                        IndexProperties indexProperties = indexSettingsValues.get(index);
+                        assert indexProperties != null
```

Comment:
Will this assert potentially cause problems in a mixed cluster?

Reply:
That's a good question. 

We read the data stream information from the master node which is either on the version that has this code or not. So, I'm tempted to say it _shouldn't_ cause problems.

Reply:
Oh no, you're right ! Great catch 👏 

So if I have a non-master node that runs on this version - say `8.11.SNAPSHOT` and the current master is on a version without this code - say `8.10` - that assertion will be tripped indeed if the `GET _data_stream` request hits the `8.11.SNAPSHOT` node. 

Will fix this tomorrow, thanks Keith !

Reply:
This has now been fixed, thanks Keith